### PR TITLE
boards: dts: Configure lpc flexcomm nodes as i2c/usart/spi at the board level 

### PR DESCRIPTION
**Pull request by MaureenHelm**
Don't assume in the soc level device trees that flexcomm nodes will
always be configured as i2c/usart/spi. Instead, configure flexcomm nodes at the
board level for lpcxpresso55s69 and lpcxpresso54114 boards.

Adds missing flexcomm nodes to lpc54xxx and lpc55s6x soc level device
trees.

### CODE REVIEW

File: dts/bindings/spi/nxp,lpc-spi.yaml [hunk 1]
```diff
@@ -5,11 +5,4 @@ description: NXP LPC SPI controller
...
 
-include: spi-controller.yaml
-
-properties:
-    reg:
-      required: true
-
-    interrupts:
-      required: true
+include: [spi-controller.yaml, "nxp,lpc-flexcomm.yaml"]
```
Comment (galak) [inline dts/bindings/spi/nxp,lpc-spi.yaml:8]: remove the quotes around "nxp,lpc-flexcomm.yaml"

File: dts/bindings/serial/nxp,lpc-usart.yaml [hunk 2]
```diff
@@ -5,11 +5,4 @@ description: LPC USART
...
 
-include: uart-controller.yaml
-
-properties:
-    reg:
-      required: true
-
-    interrupts:
-      required: true
+include: [uart-controller.yaml, "nxp,lpc-flexcomm.yaml"]
```
Comment (galak) [inline dts/bindings/serial/nxp,lpc-usart.yaml:8]: remove the quotes around "nxp,lpc-flexcomm.yaml"

File: dts/bindings/i2c/nxp,lpc-i2c.yaml [hunk 3]
```diff
@@ -5,11 +5,4 @@ description: LPC I2C node
...
 
-include: i2c-controller.yaml
-
-properties:
-    reg:
-      required: true
-
-    interrupts:
-      required: true
+include: [i2c-controller.yaml, "nxp,lpc-flexcomm.yaml"]
```
Comment (galak) [inline dts/bindings/i2c/nxp,lpc-i2c.yaml:8]: remove the quotes around "nxp,lpc-flexcomm.yaml"

File: dts/arm/nxp/nxp_lpc54xxx.dtsi [hunk 4]
```diff
@@ -12,8 +12,8 @@
 		gpio-0 = &gpio0;
 		gpio-1 = &gpio1;
 		mailbox-0 = &mailbox0;
-		spi-5 = &spi5;
-		usart-0 = &usart0;
+		spi-5 = &flexcomm5;
```
Comment (galak) [inline dts/arm/nxp/nxp_lpc54xxx.dtsi:15]: are the flexcomm's a fix function?  If not, these alias setting might be better in the board file.  (same question for the LPC55S69 aliases).

File: boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu0.dts [hunk 5]
```diff
@@ -69,11 +64,15 @@
 	status = "okay";
 };
 
-&i2c4 {
+&flexcomm0 {
```
Comment (galak) [inline boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu0.dts:67]: Seems like the place you set the flexcomm purpose is the same place you should set `status`

**Review by galak — CHANGES_REQUESTED** [review]
In general looks fine, some questions about where to put certain dt info.  Also need to fixup the quoting in the bindings.

File: dts/bindings/spi/nxp,lpc-spi.yaml (hunk 1 above)
Comment (MaureenHelm) [reply dts/bindings/spi/nxp,lpc-spi.yaml:8]: I get a device tree error without the quotes:
```
-- west build: making build dir /home/maureen/zephyrproject/zephyr/samples/sensor/fxos8700/build pristine
-- west build: generating a build system
Including boilerplate (Zephyr base): /home/maureen/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/maureen/zephyrproject/zephyr/samples/sensor/fxos8700
-- Zephyr version: 2.2.99 (/home/maureen/zephyrproject/zephyr)
-- Found PythonInterp: /usr/bin/python3.6 (found suitable version "3.6.9", minimum required is "3.6") 
-- Board: lpcxpresso55s69_cpu0
-- Found west: /home/maureen/.local/bin/west (found suitable version "0.7.2", minimum required is "0.7.1")
-- Found toolchain: zephyr (/home/maureen/tools/zephyr-sdk/0.11.1)
-- Found BOARD.dts: /home/maureen/zephyrproject/zephyr/boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu0.dts
devicetree error: 'nxp' not found
CMake Error at /home/maureen/zephyrproject/zephyr/cmake/dts.cmake:229 (message):
  gen_defines.py failed with return code: 1
Call Stack (most recent call first):
  /home/maureen/zephyrproject/zephyr/cmake/app/boilerplate.cmake:503 (include)
  /home/maureen/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /home/maureen/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:35 (include_boilerplate)
  CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /home/maureen/.local/bin/cmake -B/home/maureen/zephyrproject/zephyr/samples/sensor/fxos8700/build -S/home/maureen/zephyrproject/zephyr/samples/sensor/fxos8700 -GNinja -DBOARD=lpcxpresso55s69_cpu0 -DCONF_FILE=prj_accel.conf
```

File: dts/arm/nxp/nxp_lpc54xxx.dtsi (hunk 4 above)
Comment (MaureenHelm) [reply dts/arm/nxp/nxp_lpc54xxx.dtsi:15]: You're right, good catch. But aren't these instance aliases going away soon anyway?

File: boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu0.dts (hunk 5 above)
Comment (MaureenHelm) [reply boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu0.dts:67]: My thinking here is that the board wiring dictates the flexcomm purpose, however we have multiple cores in this soc. Therefore I set the type in the common board dtsi, but enabled it only for core 0. An application might want to use an overlay to override the default and enable it in core 1 instead.

File: dts/arm/nxp/nxp_lpc54xxx.dtsi (hunk 4 above)
Comment (galak) [reply dts/arm/nxp/nxp_lpc54xxx.dtsi:15]: They are, so you could also just remove these for now.

File: boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu0.dts (hunk 5 above)
Comment (galak) [reply boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu0.dts:67]: that makes sense.

File: dts/bindings/spi/nxp,lpc-spi.yaml (hunk 1 above)
Comment (galak) [reply dts/bindings/spi/nxp,lpc-spi.yaml:8]: guess this is due to the comma in the name.

**Review by galak — COMMENTED** [review]
So I think everything but the alias setting in the soc dtsi files is good.

File: dts/arm/nxp/nxp_lpc54xxx.dtsi (hunk 4 above)
Comment (MaureenHelm) [reply dts/arm/nxp/nxp_lpc54xxx.dtsi]: I can't remove them until the drivers are converted, but I did move them to the board file as suggested.

**Review by galak — APPROVED** [review]

**Pull request merged**